calyx-frontend/src/parser.rs

```diff
@@ -1321,24 +1360,123 @@ impl CalyxParser {
         ))
     }
 
-    fn metadata(input: Node) -> ParseResult<String> {
+    fn metadata_legacy(input: Node) -> ParseResult<String> {
         Ok(match_nodes!(input.into_children();
             [metadata_char(c)..] => c.collect::<String>().trim().into()
         ))
     }
 
+    // New Metadata
+
+    fn path_text(input: Node) -> ParseResult<PathBuf> {
+        Ok(PathBuf::from(input.as_str().trim()))
+    }
+
+    fn file_entry(input: Node) -> ParseResult<(MetadataFileId, PathBuf)> {
```

Comment:
where does the `MetadataFieldId` type come from? Should this not be `FileId`?

Reply:
This is `FileId` I just imported it with a different name since there is also
the `FileIdx` type used by the parser and the Global Position table and I didn't
want the two to get confused